### PR TITLE
Added methods to test floating point equality within tolerance

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Updated `FloatVectorTest` with test that would previously fail due to floating point rounding errors.
Problem mentioned in https://github.com/cogneco/ooc-kean/issues/870 is not fixed here, it deserves its own PR.
@marcusnaslund can you review

### CODE REVIEW

File: source/math/FloatVector2D.ooc [hunk 1]
```diff
@@ -22,7 +22,7 @@ FloatVector2D: cover {
 	x, y: Float
 	area ::= this x * this y
 	length ::= this norm
-	empty ::= !(this x > 0 && this y > 0)
+	empty ::= this y equals(0.0f) || this x equals(0.0f) || this x < 0.0f || this y < 0.0f
```
Comment (marcusnaslund) [inline source/math/FloatVector2D.ooc]: Uhm, why not `<=`?

**Comment by marcusnaslund** [comment]
I think `tolerance` is a better name than `within`.

File: source/math/FloatVector2D.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/math/FloatVector2D.ooc]: `<=` does not use the new `equals` methods and the test would fail as before.

**Comment by sebastianbaginski** [comment]
OK, renamed to `tolerance`.

File: source/math/FloatVector2D.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/math/FloatVector2D.ooc]: Alright. At least change the position of `x` and `y` in the first two statements and I'm happy.

Comment (sebastianbaginski) [inline source/math/FloatVector2D.ooc]: Done.

**Pull request merged**